madlibs.js

```diff
@@ -22,21 +22,110 @@
+      if (regex === "v") {
+        pos = "verb";
+      } else if (regex === "n") {
+        pos = "noun";
+      } else if (regex === "o") {
+        pos = "object";
+      } else if (regex === "a") {
+        pos = "adjective";
+      }
+      return { word: word, pos: pos };
+    } else {
+      return { word: ele };
+    }
+  });
+  return newArr; // new array
+}
+
+// instead of every part of speach we will place an input specifing the placeholder
+function buildStory(processedStory) {
+  const madLibsEdit = document.querySelector(".madLibsEdit");
+  const madLibsPreview = document.querySelector(".madLibsPreview");
+
+  const editParagraph = document.createElement("p");
+  const previewParagraph = document.createElement("p");
+
+  madLibsEdit.appendChild(editParagraph);
+  madLibsPreview.appendChild(previewParagraph);
+  // pass through every object inside the array
+  for (let word of processedStory) {
+    // if the word has a part of speach word
+    if (word.pos != undefined) {
+      //if its a special word with [x]
+      // create inputs
+      const inputEdit = document.createElement("input");
+      const inputPreview = document.createElement("input");
+      inputPreview.readOnly = true;
+
+      editParagraph.appendChild(inputEdit);
+      previewParagraph.appendChild(inputPreview);
+
+      inputEdit.placeholder = word.pos;
+      inputPreview.placeholder = word.pos;
+
+      inputEdit.maxLength = 20;
+
+      inputEdit.addEventListener("keyup", () => {
+        inputPreview.value = inputEdit.value;
+      });
+
+      //if the user enters ENTER key then the focus moves to the next input
+      inputEdit.addEventListener("keydown", (event) => {
+        if (event.key === "Enter" || event.key === "ArrowRight") {
+          let nextElement = inputEdit.nextElementSibling;
+          if (nextElement != null) {
+            nextElement.focus();
+          }
+        } else if (event.key === "ArrowLeft") {
+          let previousElememt = inputEdit.previousElementSibling;
+          if (previousElememt != null) {
+            previousElememt.focus();
+          }
+        }
+      });
+    } else {
+      // if it does not have part of speach then the word
+      editParagraph.append(` ${word.word}`);
+      previewParagraph.append(` ${word.word}`);
+    }
+  }
 }
 
 /**
  * All your other JavaScript code goes here, inside the function. Don't worry about
  * the `then` and `async` syntax for now.
- * 
+ *
  * You'll want to use the results of parseStory() to display the story on the page.
  */
-getRawStory().then(parseStory).then((processedStory) => {
-  console.log(processedStory);
-});
+getRawStory()
+  .then(parseStory)
+  .then((processedStory) => {
+    buildStory(processedStory);
+    const reset = document.getElementById("reset");
```

Comment:
You could add the code below into a rest function and call it here to make the code cleaner.